C/services/notification/notification_api.cpp

```diff
@@ -739,6 +797,48 @@ bool NotificationApi::queueStatsRateNotification(const string& statistic,
 	return queue->addElement(item);
 }
 
+/**
+ * Add alert data of asset name into the process queue
+ *
+ * @param payload	The data for the audit code
+ * @return		false error, true on success
+ */
+bool NotificationApi::queueAlertNotification(const string& payload)
+{
+	Logger::getLogger()->debug("Recieved alert notification: %s", payload.c_str());
+
+	Reading *reading = new Reading("alert", payload);
+	vector<Reading *> readingVec;
+	readingVec.push_back(reading);
+	ReadingSet* readings = NULL;
+	try
+	{
+		readings = new ReadingSet(&readingVec);
+	}
+	catch (exception* ex)
```

Comment:
Constructor of ReadingSet thows exception of type ReadingSetException. Please catch by ReadingSetException* to align with  ReadingSet class